
syntax = "proto3";

package hello;

option go_package="./scrape_client/scrapepb";

message Product {
  string name = 1;
  string url = 2;
  string price = 3;
  string img_url = 4;
  repeated string region_list = 5;
}

message ScrapeManyTimesRequest {
  string productName = 1;
  float userLat = 2;
  float userLon = 3;
}

message ScrapeManyTimesResponse {
  Product product = 1;
  string storeName = 2;
  float storeLat = 3;
  float storeLon = 4;
}


service ScrapingService {
  rpc ScrapeManyTimes (ScrapeManyTimesRequest) returns (stream ScrapeManyTimesResponse) {}
}